### PR TITLE
Introducing an RFC process for the RabbitMQ Project

### PR DESCRIPTION
**Pull request by dumbbell**
The goal is to make the design of new/impactful changes open, so the community can participate by submitting or giving feedback to RFCs.

### CODE REVIEW

**Comment by dumbbell** [comment]
As mentionned in the RFC text, the proposed process is [described in detail in an external gist](https://gist.github.com/dumbbell/b59ec82d57b246f68916e42fd8b7f837), along with a template for future RFCs.

Even though the content of this gist may be added to this `rfcs` repository in the end, the process of accepting the fact we want to adopt an RFC process (this PR which is the RFC of the RFC process) and the submission of the process documentation and the RFC template (the implementation) are two separate things.

So basically, the workflow will be:
1. We discuss that we want or not an RFC process with all the details of that process (using this PR)
2. We submit a second PR to implement it (the README.md describing the process + the RFC template).

Yes, this looks a bit awkward for this very specific case, but this is to experiment the entire process.

**Comment by lukebakken** [comment]
I would like to see GitHub labels created to indicate where in the RFC workflow a particular PR or Issue is.

**Comment by dumbbell** [comment]
Yes, this is a good idea. Here are the one used during the Rust RFC process:

* `A-$area` where `$area` categorizes the RFC.
* `T-$team` because they have several sub-teams.
* `proposed-final-comment-period` to indicate the Core team should sign off the RFC to enter the *final comment period*.
* `final-comment-period` (along with the `disposition-*`) to indicate the RFC is in the *final comment period*, lasting 10 days.
* `finished-final-comment-period` to indicate the *final comment period* is over.
* `disposition-{merge,postpone,close}` when entering the *final comment period* to notify the decision of the Core team.
* `closed` to indicate an RFC which was closed and rejected.
* `postponed` to indicate an RFC which was closed but can be revisited in the future.

**Comment by michaelklishin** [comment]
This looks like a good start to me.
